core/src/main/java/org/elasticsearch/index/analysis/IndexAnalyzers.java

```diff
@@ -0,0 +1,106 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.elasticsearch.index.analysis;
+
+import org.elasticsearch.index.AbstractIndexComponent;
+import org.elasticsearch.index.IndexSettings;
+
+import java.io.Closeable;
+import java.util.Map;
+
+/**
+ * IndexAnalyzers contains a name to analyzer mapping for a specific index.
+ * This class only holds analyzers that are explicitly configured for an index and doesn't allow
+ * access to individual tokenizers, char or token filter.
+ *
+ * @see AnalysisRegistry
+ */
+public final class IndexAnalyzers extends AbstractIndexComponent implements Closeable {
+    private final NamedAnalyzer defaultIndexAnalyzer;
+    private final NamedAnalyzer defaultSearchAnalyzer;
+    private final NamedAnalyzer defaultSearchQuoteAnalyzer;
+    private final Map<String, NamedAnalyzer> analyzers;
+    private final IndexSettings indexSettings;
+
+    public IndexAnalyzers(IndexSettings indexSettings, NamedAnalyzer defaultIndexAnalyzer, NamedAnalyzer defaultSearchAnalyzer,
+                          NamedAnalyzer defaultSearchQuoteAnalyzer, Map<String, NamedAnalyzer> analyzers) {
+        super(indexSettings);
+        this.defaultIndexAnalyzer = defaultIndexAnalyzer;
+        this.defaultSearchAnalyzer = defaultSearchAnalyzer;
+        this.defaultSearchQuoteAnalyzer = defaultSearchQuoteAnalyzer;
+        this.analyzers = analyzers;
+        this.indexSettings = indexSettings;
+    }
+
+    /**
+     * Returns an analyzer mapped to the given name or <code>null</code> if not present
+     */
+    public NamedAnalyzer get(String name) {
+        return analyzers.get(name);
+    }
+
+
+    /**
+     * Returns the default index analyzer for this index
+     */
+    public NamedAnalyzer getDefaultIndexAnalyzer() {
+        return defaultIndexAnalyzer;
+    }
+
+    /**
+     * Returns the default search analyzer for this index
+     */
+    public NamedAnalyzer getDefaultSearchAnalyzer() {
+        return defaultSearchAnalyzer;
+    }
+
+    /**
+     * Returns the default search quote analyzer for this index
+     */
+    public NamedAnalyzer getDefaultSearchQuoteAnalyzer() {
+        return defaultSearchQuoteAnalyzer;
+    }
+
+    @Override
+    public void close() {
+        for (NamedAnalyzer analyzer : analyzers.values()) {
+            if (analyzer.scope() == AnalyzerScope.INDEX) {
+                try {
+                    analyzer.close();
+                } catch (NullPointerException e) {
+                    // because analyzers are aliased, they might be closed several times
```

Comment:
If we do need to keep this clause can you add an assertion that the index's version created setting is less than 5.0?